### PR TITLE
feat(Chat): save draft messages

### PR DESCRIPTION
**Pull request by juliahermak**
- [x] Add save draft messages functionality
- [x] Rewrite to TS
- [x] Get rid of `this.$route.query.replyToId`

### CODE REVIEW

**Comment by adamant-al** [comment]
One more thing, keep reply/quote state.

**Comment by bludnic** [comment]
[Feat: Save texts in input fields—message drafts](https://trello.com/c/am4oBJ6s/263-feat-save-texts-in-input-fields-message-drafts)

**Comment by juliahermak** [comment]
@adamant-al look now

**Review by adamant-al — APPROVED** [review]
Works well.
Looks good.

**Comment by juliahermak** [comment]
Added.

File: src/components/Chat/Chat.vue [hunk 1]
```diff
@@ -484,8 +486,8 @@ export default {
         'Windows 10': this.$t('chats.message_windows_10')
       }[detect().os] || this.$t('chats.message')
 
-    if (this.$route.query.replyToId) {
-      this.replyMessageId = this.$route.query.replyToId
+    if (this.$store.getters['draftMessage/draftReplyTold'](this.partnerId)) {
+      this.replyMessageId = this.$store.getters['draftMessage/draftReplyTold'](this.partnerId)
```
Comment (bludnic) [inline src/components/Chat/Chat.vue:490]:
```suggestion
   const draftMessage = this.$store.getters['draftMessage/draftReplyTold'](this.partnerId)
    if (draftMessage) {
      this.replyMessageId = draftMessage
```

File: src/components/Chat/Chat.vue [hunk 2]
```diff
@@ -510,7 +512,15 @@ export default {
...
     },
+    cancelReplyMessage() {
+      this.replyMessageId = -1
+      this.$store.commit('draftMessage/deleteReplyTold', {
+        replyToId: this.replyMessageId,
+        partnerId: this.partnerId
+      })
+    },
     sendMessage(message) {
+      this.$store.dispatch('draftMessage/deleteWhenSend', { partnerId: this.partnerId })
```
Comment (bludnic) [inline src/components/Chat/Chat.vue:523]: draftMessage/removeDraft
draftMessage/deleteDraft

File: src/store/types.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { DraftState } from '@/store/modules/draft-message'
```
Comment (bludnic) [inline src/store/types.ts:1]: Enter

File: src/store/index.js [hunk 4]
```diff
@@ -133,6 +134,7 @@ const store = {
     },
     logout({ dispatch }) {
       dispatch('reset')
+      dispatch('draftMessage/resetState', null, { root: true })
```
Comment (bludnic) [inline src/store/index.js:137]: Delete

File: src/store/modules/draft-message/index.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    } else {
+      return ''
+    }
+  }
+}
+
+const actions: ActionTree<DraftState, RootState> = {
+  resetState(context) {
+    context.commit('reset')
+  },
```
Comment (bludnic) [inline src/store/modules/draft-message/index.ts:69]: Delete

**Review by bludnic — CHANGES_REQUESTED** [review]

**Review by bludnic — APPROVED** [review]

**Pull request merged**